main/SQLHelper.h

```diff
@@ -226,13 +228,15 @@ struct _tTaskItem
 			getclock(&tItem._DelayTimeBegin);
 		return tItem;
 	}
-	static _tTaskItem SetSetPoint(const float DelayTime, const uint64_t idx, const std::string &varvalue)
+	static _tTaskItem SetSetPoint(const float DelayTime, const uint64_t idx, const std::string &varvalue, const std::string &mode = std::string(), const std::string &until = std::string())
 	{
 		_tTaskItem tItem;
 		tItem._ItemType = TITEM_SET_SETPOINT;
 		tItem._DelayTime = DelayTime;
 		tItem._idx = idx;
 		tItem._sValue = varvalue;
+		tItem._sMode = mode;
```

Comment:
Your solution is OK. But maybe this could have been set in the sValue delimited with a semicolon ( ; )
This because in the feature i can see 100 new variables being added here...
Is it possible to change this ?

Reply:
I'm reluctant to break the string apart when it comes in from the script, then put it back together again, then break it apart again. I suppose we could just pass the whole "temp#mode#until" string through and then break it apart in `CSQLHelpers::Do_Work()` but then we lose out on the error handling when the arguments are bogus. 

In the general case though... this is what unions are for ;)

Reply:
I could also just abuse the `_command` field (like `TITEM_SEND_NOTIFICATION` does) and the `_relatedEvent` field...

Reply:
Hm, not `_relatedEvent` as that gets interpreted generically, but `_command` seems to be fair game. So now I'm adding only *one* strnig. Is that an acceptable compromise?